### PR TITLE
Pvc volume to avoid redownloading compilers

### PR DESCRIPTION
**Pull request by eruizgar91**
### Description

The purpose of this is avoid re-download the compilers when there is a new release. 
It's installing a nfs server to allow multiple read-write.

### Tested

Tested

### Related issues

- Fixes https://github.com/celo-org/data-services/issues/179

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/helm-charts/blockscout/values.yaml [hunk 1]
```diff
@@ -189,3 +189,10 @@ blockscout:
     image:
       repository: gcr.io/celo-testnet/celo-monorepo
       tag: metadata-crawler-77a392216d4927e85ce4b683508fc0539aa92a34
+nfs-server-provisioner:
+  persistence:
+    enabled: true
+    storageClass: "ssd"
+    size: 20Gi
```
Comment (jcortejoso) [inline packages/helm-charts/blockscout/values.yaml]: This is used for `PVC Storage Request for config volume`. I'd say probably 1 or 2 Gb would be much than enough.

File: packages/helm-charts/blockscout/templates/_helpers.tpl [hunk 2]
```diff
@@ -99,10 +99,15 @@ the `volumes` section.
...
       readOnly: true
+    - name: vyper-compilers
+      mountPath: /opt/app/apps/explorer/priv/vyper_compilers
 volumes:
   - name: blockscout-cloudsql-credentials
     secret:
       secretName: blockscout-cloudsql-credentials
+  - name: vyper-compilers
+    persistentVolumeClaim:
+      claimName: nfs-volume
```
Comment (jcortejoso) [inline packages/helm-charts/blockscout/templates/_helpers.tpl:110]: Where is this PVC created?

**Comment by jcortejoso** [comment]
Current approach would conflict in clusters with more than one blockscout deployment. We'd need to think if we want to decouple the `nfs-server-provisioner` from blockscout deployment, or if we want to customize any current common object (at least the storageClass) to be different for each deployment.

**Comment by carterqw2** [comment]
@jcortejoso can we maybe have a global volume? This is really a storage for compilers, it can be easily reused by all of the deployments.

**Review by alvarof2 — APPROVED** [review]

**Comment by jcortejoso** [comment]
@[carterqw2](https://github.com/carterqw2) @eruizgar91 , @alvarof2  me was discussing about this. Although a global volume would be something possible and achievable, the options we have thing about to implement it have some drawbacks that can make them less suitable:
- Having a shared NFS volume across all clusters. This would be somehow similar to the implementation from this PR, but instead of having different NFS volumes per cluster, there would be one unique volume. The main difficulty for this approach is how you manage the network connection to the volume endpoint in a secure way. NFS is tricky to secure and traditionally the security is based on network restrictions (based on local networks, or security groups/ips). For this case the two options I can think about are a) expose the NFS service to a local ip and setup ssh tunnels/vpns across the clusters to access the NFS endpoint, or b) expose the NFS endpoint to a public ip and control the access using firewall rules, but we'd need to know the outgoing ips that the cluster would use to connect to the Internet (and keep them updated). Both of them add some complexity or POF to the setup so we'd need to consider carefully.
- Second option is using a NFS-kind service like GCP Filestore, but we discard this because the minimum size you can allocate is 1TB per volume, and the costs are considerable high for the needs here.

If we think there is a strong argument to keep considering a unique volume for all cluster, I'd explore the options for mounting file repositories like GCS or S3 as filesystems using Kubernetes volumes (also considering performance is probably not a key factor here). This could be interesting to investigate but we'd need to spend some more time checking the options and confirming there is a solution that could work for the case.

File: packages/celotool/src/lib/cluster.ts [hunk 3]
```diff
@@ -137,6 +138,8 @@ export async function setupCluster(celoEnv: string, createdCluster: boolean) {
 
   await installCertManagerAndNginx(celoEnv)
 
+  await installNFSServerProvisioner(celoEnv)
```
Comment (jcortejoso) [inline packages/celotool/src/lib/cluster.ts]: I'm thinking now if this would be more appropriate to move to blockscout install/upgrade command/logic instead of cluster setup. My argument for moving is not installing the NFS chart on clusters that won't run blockscout (like oracle's or forno's), bc right now the NFS dependency is very specific to blockscout.
My impression is moving this function invocation to the blosckout deployment commands should work as it is now.

File: packages/celotool/src/lib/k8s-cluster/base.ts [hunk 4]
```diff
@@ -83,6 +84,7 @@ export abstract class BaseClusterManager {
 
       await installCertManagerAndNginx(this.celoEnv, this.clusterConfig)
       await installAndEnableMetricsDeps(true, context, this.clusterConfig)
+      await installNFSServerProvisioner(this.celoEnv)
```
Comment (jcortejoso) [inline packages/celotool/src/lib/k8s-cluster/base.ts]: same as before

**Review by jcortejoso — CHANGES_REQUESTED** [review]

**Review by jcortejoso — APPROVED** [review]
looking good

File: packages/helm-charts/blockscout/values.yaml [hunk 5]
```diff
@@ -163,7 +163,7 @@ blockscout:
       walletList: '[{"url":"https://valoraapp.com/", "title":"Valora"}, {"url":"https://celoterminal.com/", "title":"Celo Terminal"}, {"url":"https://celowallet.app/", "title":"Celo Wallet"}, {"url":"https://www.nodewallet.xyz/", "title":"Node Wallet"}]'
       nftList: '[{"url":"https://niftydrop.net/", "title":"Niftydrop"}, {"url":"https://nfts.valoraapp.com/", "title":"NFT Viewer"}, {"url":"https://cyberbox.art/", "title":"Cyberbox"}, {"url":"https://nom.space/", "title":"Nomspace"}, {"url":"https://alities.io/", "title":"Alities"}]'
       connectList: '[{"url":"https://impactmarket.com/", "title":"impactMarket"}, {"url":"https://talentprotocol.com/", "title":"Talent Protocol"}, {"url":"https://doni.app/", "title":"Doni"}]'
-      spendList: '[{"url":"https://giftcards.bidali.com/spend-celo-dollarce", "title":"Bidali"}, {"url":"https://flywallet.io/", "title":"Flywallet"},{"url":"https://chispend.com/imp", "title":"ChiSpend"}]'
+      spendList: '[{"url":"https://giftcards.bidali.com/", "title":"Bidali"}, {"url":"https://flywallet.io/", "title":"Flywallet"},{"url":"https://chispend.com/", "title":"ChiSpend"}]'
```
Comment (carterqw2) [inline packages/helm-charts/blockscout/values.yaml]: Let's move it to a separate PR (it needs to be merged and deployed sooner).
